### PR TITLE
rootless: add the built-in slirp DNS server

### PR DESCRIPTION
**Pull request by giuseppe**
add the slirp built-in DNS server when using slirp4netns.

Closes: https://github.com/containers/buildah/issues/1660

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: run_linux.go [hunk 1]
```diff
@@ -479,6 +479,15 @@ func (b *Builder) addNetworkConfig(rdir, hostPath string, chownOpts *idtools.IDP
 	if len(dnsSearch) > 0 {
 		search = dnsSearch
 	}
+
+	if b.Isolation == IsolationOCIRootless {
+		ns := namespaceOptions.Find(string(specs.NetworkNamespace))
+		if ns != nil && !ns.Host && ns.Path == "" {
+			// if we are using slirp4netns, also add the built-in DNS server.
+			nameservers = append([]string{"10.0.2.3"}, nameservers...)
```
Comment (rhatdan) [inline run_linux.go]: Can you at least log that we are doing this.  Is it possible to add the comment line here as well like you did in podman?

**Comment by TomSweeneyRedHat** [comment]
Test error looks to be a network flake.

File: run_linux.go (hunk 1 above)
Comment (giuseppe) [reply run_linux.go]: I've added a debug log message

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM

**Comment by giuseppe** [comment]
@TomSweeneyRedHat does it look fine now?

**Comment by TomSweeneyRedHat** [comment]
LGTM and sorry for the slow response @giuseppe I must have missed the updates from GitHub on this one.

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**